Refresh

pjhartzell opened the pull request:
Brings the repo up to date with the current `stactools-packages` template contents.

Edits to existing source and test code limited to:
-  Changes imparted by the formatters (`isort`, `yapf`, etc.)
- A few minor changes to address `mypy` complaints in `utils.py` and `mtl_metadata.py`

pjhartzell committed:
update stactools to >= 0.2.6, remove Python 3.6 classifier

pjhartzell committed:
remove setup.py

pjhartzell committed:
update to use pytest

pjhartzell committed:
update requirements-dev.txt contents

pjhartzell committed:
update format script to include isort

pjhartzell committed:
update lint script

- add mypy
- add .editorconfig

pjhartzell committed:
update READMEs, LICENSE, .gitignore, requirements-dev

pjhartzell committed:
add docker files

pjhartzell committed:
update docker and script folder content

pjhartzell committed:
add docs folder

pjhartzell committed:
update .github/ with new templates and workflows

gadomski commented on .github/PULL_REQUEST_TEMPLATE.MD:
```diff
@@ -0,0 +1,15 @@
+- [ ] Code lints properly (run `scripts/lint`).
+- [ ] Tests pass (run `scripts/test`).
+- [ ] Documentation has been updated to reflect changes, if applicable.
+- [ ] Changes are added to the [CHANGELOG](../CHANGELOG.md).
```
```suggestion
- [ ] Changes are added to the [CHANGELOG](../CHANGELOG.md).
- [ ] Example STAC JSON files are updated in `examples/` (using `scripts/create_expected`)
```

pjhartzell replied:
added

gadomski commented on CHANGELOG.MD:
```diff
@@ -4,8 +4,20 @@ All notable changes to this project will be documented in this file.
-## [v0.2.0]
+## [Unreleased]
 
-### Changed
+### Added
 
-- Upgrade to stactools 0.2.1.a2 (supporting PySTAC 1.0.0)
+- Nothing.
+
+### Deprecated
+
+- Nothing.
+
+### Removed
+
+- Nothing.
+
+### Fixed
+
+- Nothing.
```
We should keep the CHANGLOG information, but simply update it w/ an `Unreleased` section and links, e.g. https://github.com/stactools-packages/noaa-c-cap/blob/main/CHANGELOG.md.

pjhartzell replied:
Yeah. Done.

gadomski commented on LICENSE:
```diff
@@ -1,6 +1,6 @@
 This software is licensed under the Apache 2 license, quoted below.
 
-Copyright 2021 COMPANY [COMPANY WEBPAGE URL]
+Copyright 2022 COMPANY [COMPANY WEBPAGE URL]
```
We should probably update this to Microsoft's, e.g. https://github.com/stactools-packages/noaa-c-cap/blob/main/LICENSE#L3.

pjhartzell replied:
Done.

gadomski commented on README.md:
```diff
@@ -1,15 +1,18 @@
 
-1. Clone this repository and name it `stactools-{NAME}`, where `NAME` is your package name.
+1. Clone this template repository as your package name, e.g. `landsat`.
    This name should be short, memorable, and a valid Python package name (i.e. it shouldn't start with a number, etc).
-2. Update `setup.cfg` with your package name, description, and such.
-3. Rename `src/stactools.landsat` to `src/stactools/{NAME}`.
-4. Rewrite this README to provide information about how to use your package.
-5. Update the LICENSE with your company's information (or whomever holds the copyright).
-6. Update the environment name in `environment.yml`.
-7. Update the environment variables in `.github/workflows/release.yml` to the appropriate values to publish for your organization.
-8. Update all scripts in the `docker` directory to refer to `stactools-{NAME}` and `stactools-{NAME}-dev`.
+   It can, however, include a hyphen, in which case the name for Python imports will be the underscored version, e.g. `landsat-8` goes to `stactools.landsat_8`.
+   Your name will be used on PyPI to publish the package in the stactools namespace, e.g. `stactools-landsat`.
+2. Change into the top-level directory of your package and run `scripts/rename`.
+   This will update _most_ of the files in the repository with your new package name.
+3. Update `setup.cfg` with your package description and such.
+4. Update the LICENSE with your company's information (or whomever holds the copyright).
+5. Run `sphinx-quickstart` in the `docs` directory to create the documentation template.
+6. Update `docs/installation_and_basic_usage.ipynb` to provide an interactive notebook to help users get started. Include the following badge at the top of the README to launch the notebook: [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/stactools-packages/template/main?filepath=docs/installation_and_basic_usage.ipynb). Be sure to modify the badge href to match your p ...
+7. Add example Items (and Collections and Catalogs, if included) to a `examples/` directory.
+8. Delete this file, and rename `README-template.md` to `README.md`. Update your new README to provide information about how to use your package.
```
The README should be updated to the information from `README-tempalte`.

pjhartzell replied:
Done.

gadomski commented on README-template.md:
```diff
@@ -0,0 +1,32 @@
+[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/stactools-packages/ephemeral/main?filepath=docs/installation_and_basic_usage.ipynb)
```
The content of this file should be copied/updated to README.md and then this file should be deleted.

pjhartzell replied:
Done.

pjhartzell committed:
correct publish script usage text

pjhartzell committed:
address PR review comments:

- proper README
- retain CHANGELOG content
- update LICENSE
- add docs with working example in notebook
- add example item to examples folder
- update setup.cfg

pjhartzell committed:
fix linting errors

gadomski commented on CHANGELOG.MD:
```diff
@@ -4,8 +4,19 @@ All notable changes to this project will be documented in this file.
 
 - Upgrade to stactools 0.2.1.a2 (supporting PySTAC 1.0.0)
+
+[0.2.0]: <https://github.com/stactools-packages/landsat/releases/tag/v0.2.0>
```
```suggestion
[Unreleased]: <https://github.com/stactools-packages/landsat/compare/v0.2.0..main>
[0.2.0]: <https://github.com/stactools-packages/landsat/releases/tag/v0.2.0>
```

pjhartzell replied:
Added. Cool stuff. Was not aware of releases and comparison links.

gadomski commented on README.md:
````diff
@@ -1,15 +1,30 @@
 
-## How to use
+```bash
+$ stac landsat create-item "/tests/data-files/assets/LC08_L2SP_005009_20150710_20200908_02_T2_MTL.xml" "examples/item/LC08_L2SP_005009_20150710_02_T2.json"
````
```suggestion
$ stac landsat create-item "tests/data-files/assets/LC08_L2SP_005009_20150710_20200908_02_T2_MTL.xml" "examples/item/LC08_L2SP_005009_20150710_02_T2.json"
```

pjhartzell replied:
Fixed.

gadomski committed:
docs: add (empty) _static dir

pjhartzell committed:
chore: edits and updates

- add unreleased comparison link
- fix example cli command
- update to latest changes in stactools template

gadomski commented on CHANGELOG.MD:
```diff
@@ -4,8 +4,20 @@ All notable changes to this project will be documented in this file.
+### Updated
+
+- Repo brought up to date with the current stactools-packages template contents as of 2022-02-18.
+- Edits to existing source and test code as part of the update were limited to:
+    - Format changes imparted by the `isort`, `yapf`, `flake8`, and `EditorConfig`
+    - A few minor changes to address `mypy` complaints in `utils.py` and `mtl_metadata.py`
```
So I'd probably cut this section from the CHANGELOG, since we will likely be including actual source and test code edits before the next release.

pjhartzell replied:
Sounds good. Done.

gadomski reviewed (changes requested):
Just one changelog quibble and then should be good to :shipit:.

pjhartzell committed:
remove unneccesary text from CHANGELOG
